xbmc/input/InputCodingTableBaiduPY.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+*      Copyright (C) 2005-2013 Team XBMC
+*      http://xbmc.org
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with XBMC; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <stdlib.h>
+#include "InputCodingTableBaiduPY.h"
+#include "filesystem/CurlFile.h"
+#include "utils/StringUtils.h"
+#include "utils/RegExp.h"
+#include "guilib/GUIMessage.h"
+#include "guilib/GUIWindowManager.h"
+
+CInputCodingTableBaiduPY::CInputCodingTableBaiduPY() :
+  CThread("BaiduPYApi"),
+  m_messageCounter{ 0 }
+{
+  m_codechars = "abcdefghijklmnopqrstuvwxyz";
+  m_code = "";
+  Create();
+}
+
+void CInputCodingTableBaiduPY::Process()
+{
+  while (!m_bStop) //Make sure we don't exit the thread
+  {
+    AbortableWait(m_Event, -1); //Wait for work to appear
+    while (!m_bStop) //Process all queued work before going back to wait on the event
+    {
+      CSingleLock lock(m_CS);
+      if (m_work.empty())
+        break;
+
+      auto work = m_work.front();
+      m_work.pop_front();
+      lock.Leave();
+
+      std::string data;
+      XFILE::CCurlFile http;
+      std::string strUrl;
+      strUrl = StringUtils::Format("http://olime.baidu.com/py?input=%s&inputtype=py&bg=%d&ed=%d&result=hanzi&resultcoding=unicode&ch_en=0&clientinfo=web", work.c_str(), m_api_begin, m_api_end);
+
+      if (http.Get(strUrl, data))
+      {
+        HandleResponse(work, data);
+      }
+    }
+  }
+}
+
+void CInputCodingTableBaiduPY::HandleResponse(const std::string& strCode, const std::string& response)
+{
+  if (strCode != m_code) // don't handle obsolete response
+    return;
+
+  auto words = std::make_shared<std::vector<std::wstring>>();

```

Comment:
get rid of the shared_ptr, just use a local std::vector since it'll get copied/moved into the map before we exit the method